File: ios/Theme/General.swift
import Foundation

public class General {
  private let style: Style

  init() {
    self.style = Style()
  }

  public func getBorderRadius(_ theme: NSDictionary?, key: String) -> Int32 {
    let defaultBorderRadius: Int32 = 20
    if !self.style.exists(theme, key: key) {
      return defaultBorderRadius
    }

    let borderRadius = (theme?[key] as? Int32) ?? defaultBorderRadius
    return borderRadius < 0 ? defaultBorderRadius : borderRadius
  }
  
  public func getBorderRadius(_ theme: NSDictionary?, key: String, defaultBorderRadius: Int32) -> Int32 {
    if !self.style.exists(theme, key: key) {
      return defaultBorderRadius
    }

    let borderRadius = (theme?[key] as? Int32) ?? defaultBorderRadius
    return borderRadius < 0 ? defaultBorderRadius : borderRadius
  }

  public func getStatusBarStyle(_ key: String) -> UIStatusBarStyle {
    var defaultStatusBarStyle = UIStatusBarStyle.default
    if #available(iOS 13, *) {
      defaultStatusBarStyle = UIStatusBarStyle.darkContent
    }

    if !self.style.exists(key) {
      return defaultStatusBarStyle
    }

    let statusBarColor = (Theme.Style?[key] as? String) ?? ""
    if statusBarColor.isEmpty {
      return defaultStatusBarStyle
    }

    if #available(iOS 13, *) {
      switch statusBarColor {
      case "DARK_CONTENT":
        do {
          return defaultStatusBarStyle
        }
      case "LIGHT_CONTENT":
        do {
          return UIStatusBarStyle.lightContent
        }
      case "DEFAULT":
        do {
          return UIStatusBarStyle.default
        }
      default:
        do {
          return defaultStatusBarStyle
        }
      }
    }

    return defaultStatusBarStyle
  }
}

File: examples/flutterapp/android/app/src/main/java/com/azifacemobile/theme/General.java
package com.azifacemobile.theme;

import com.azifacemobile.utils.Theme;
import com.facetec.sdk.FaceTecCancelButtonCustomization.ButtonLocation;

import org.json.JSONException;
import org.json.JSONObject;

public class General {
  private final Theme theme;

  public General() {
    this.theme = new Theme();
  }

  public int getBorderRadius(JSONObject theme, String key) {
    final int defaultBorderRadius = 20;
    try {
      if (!this.theme.exists(theme, key)) {
        return defaultBorderRadius;
      }

      final int borderRadius = theme.getInt(key);
      return borderRadius < 0 ? defaultBorderRadius : borderRadius;
    } catch (JSONException e) {
      return defaultBorderRadius;
    }
  }

  public ButtonLocation getButtonLocation(String key) {
    final ButtonLocation defaultLocation = ButtonLocation.TOP_RIGHT;
    if (!this.theme.exists(key)) {
      return defaultLocation;
    }
    Object value = com.azifacemobile.theme.Theme.Style.get(key);
    final String buttonLocation = (value instanceof String) ? (String) value : null;
    if (buttonLocation == null) {
      return defaultLocation;
    }
    if (buttonLocation.isEmpty()) {
      return defaultLocation;
    }

    switch (buttonLocation) {
      case "TOP_RIGHT":
        return ButtonLocation.TOP_RIGHT;
      case "TOP_LEFT":
        return ButtonLocation.TOP_LEFT;
      case "DISABLED":
        return ButtonLocation.DISABLED;
      default:
        return defaultLocation;
    }
  }
}
